transformer/transformer.go

```diff
@@ -0,0 +1,289 @@
+package transformer
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const startHighlightTag string = "<strong>"
+const endHighlightTag string = "</strong>"
+
+// Transformer represents an instance of the ResposeTransformer interface
+type Transformer struct {
+}
+
+type searchResponse struct {
+	Count        int           `json:"count"`
+	Took         int           `json:"took"`
+	Items        []contentItem `json:"items"`
+	ContentTypes []contentType `json:"content_types"`
+}
+type contentItem struct {
+	Description description `json:"description"`
+	Type        string      `json:"type"`
+	URI         string      `json:"uri"`
+	Matches     matches     `json:"matches,omitempty"`
+}
+type description struct {
+	Summary           string   `json:"summary"`
+	NextRelease       string   `json:"next_release,omitempty"`
+	Unit              string   `json:"unit,omitempty"`
+	Keywords          []string `json:"keywords,omitempty"`
+	ReleaseDate       string   `json:"release_date,omitempty"`
+	Edition           string   `json:"edition,omitempty"`
+	LatestRelease     bool     `json:"latest_release,omitempty"`
+	Language          string   `json:"language,omitempty"`
+	Contact           contact  `json:"contact,omitempty"`
+	DatasetID         string   `json:"dataset_id,omitempty"`
+	Source            string   `json:"source,omitempty"`
+	Title             string   `json:"title"`
+	MetaDescription   string   `json:"meta_description,omitempty"`
+	NationalStatistic bool     `json:"national_statistic,omitempty"`
+	Headline1         string   `json:"headline1,omitempty"`
+	Headline2         string   `json:"headline2,omitempty"`
+	Headline3         string   `json:"headline3,omitempty"`
+}
+type contact struct {
+	Name      string `json:"name,omitempty"`
+	Telephone string `json:"telephone,omitempty"`
+	Email     string `json:"email,omitempty"`
+}
+type matches struct {
+	Description struct {
+		Summary         []matchDetails `json:"summary"`
```

Comment:
Summary and Title are optional in swagger spec

Reply:
Code updated to make optional fields pointers